check_installed_programs.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+if [ $# -gt 0 ]; then
+    programs=("$@")
+else
+    read -ra programs
+fi
+
```

Comment:
Can you add a check (with a descriptive error) for ensuring whatever nix commands you use in this script are actually available? Essentially, stop users from running this if they don't have nix installed, and tell them why